azure-pipelines-scheduled.yaml

```diff
@@ -0,0 +1,100 @@
+name: HMCTS Management Information ADF Self Hosted Integration Runtime Image - Scheduled
+
+schedules:
+  - cron: "0 22 * * *"
+    displayName: Every day at 10pm
+    branches:
+      include:
+        - master
+    always: true
+
+trigger: none
+pr: none
+
+pool:
+  vmImage: 'windows-2019'
+
+parameters:
+  - name: environment
+    displayName: Environment
+    type: string
+    default: 'dev'
+    values:
+      - 'sbox'
+      - 'dev'
+      - 'test'
+      - 'ithc'
+      - 'stg'
+      - 'prod'
+
+variables:
+  projectName: 'mi'
+  applicationName: 'adf-integration-runtime'
+  ${{ if eq(variables['Build.SourceBranchName'], 'master') }}:
+    environment: 'prod'
+  ${{ if not(eq(variables['Build.SourceBranchName'], 'master')) }}:
+    environment: ${{ parameters.environment }}
+  azureSubscriptionEndpoint: DTS-SHAREDSERVICES-${{ upper(variables['environment']) }}
+  azureContainerRegistry: 'sdshmctspublic.azurecr.io'
+
+stages:
+  - stage: CheckShirVersion
+    jobs:
+      - job: CheckShirVersion
+        steps:
+          - task: AzureCLI@2
+            displayName: 'Get current Self-hosted IR version'
+            name: 'getSelfHostedIrVersion'
+            inputs:
+              azureSubscription: $(azureSubscriptionEndpoint)
+              scriptType: bash
+              scriptLocation: inlineScript
+              inlineScript: |
+                az config set extension.use_dynamic_install=yes_without_prompt
+                current_version=$(az datafactory integration-runtime get-status --factory-name "mi-ingestion-adf-$(environment)" --name "IR-Self-Hosted" --resource-group "mi-$(environment)-rg" --query properties.version | tr -d '"')
```

Comment:
Logic is:

Check current SHIR version reported by the ADF.
If different from download link version, set UP_TO_DATE variable as false so build image will be done.

Note this does mean if the ADF SHIR version is newer than the download link version, it will downgrade, but its better to maintain the download link version as usually thats the current stable version.